src/Sylius/Behat/Context/Api/Shop/ProductContext.php

```diff
@@ -662,4 +676,31 @@ private function hasProductWithMainImage(): bool
 
         return $images[0]['type'] === 'main' && $images[0]['path'];
     }
+
+    private function productHasAssociationsWithProducts(array $associationsIris, string $productAssociationName, array $products): bool
+    {
+        foreach ($associationsIris as $associationIri) {
+            $associationResponse = $this->client->showByIri($associationIri);
+            $associationTypeIri = $this->responseChecker->getValue($associationResponse, 'type');
+            $associationTypeResponse = $this->client->showByIri($associationTypeIri);
+
+            if ($this->responseChecker->hasValue($associationTypeResponse, 'name', $productAssociationName)) {
```

Comment:
It can be refactored in such a way that providing the association type for a given name could be moved to a separate private method. But I would be for doing that in a separate PR 😃